### ISSUE TITLE
Clearing out a line?

### ISSUE THREAD

**Issue by rjstanford** [teodoran/d3-knockout-demo#1]
Great approach!  I'm new to D3 but familiar with Knockout.  Can you give me some advice on how to clear out a line chart to display new content in it?  It appears that whatever happens to the array being passed in, all of the data is added to the chart instead of replacing the data that already exists. I see that that has some advantages for adding new points, but it makes it difficult to display a variety of different data.

Currently if you pass in an empty array you get a D3 error.  I've tried doing a variety of things like:

``` javascript
        if (data == null) {
            svg.select("path.line.data").remove();
            svg.append("path").attr("class", "line data");
            return;
        }
```

but if I do remove the line it never adds new data (oddly enough it will continue to adjust the axis).

Any advice? Thanks again...

**Comment by magnuskiro** [comment]
I think that the answer you are looking for is in example1(e1.html, election-data-view-model.js, election-binding.js). Where we in the view-model updates the date of the graph at an interval. 
This is a barChart example, but I think changing out the data variable should do the trick.
You can also add dummy data for instantiation of the variable to prevent d3 to give an error, points like 0,0 would not show with the user. 

The code in question(from election-data-view-model.js): 

```
    getNum = function () {
            return Math.random() * 100
            //return 1;
        };

    self.electionData = ko.observable(37);

    setInterval(function () {
        self.electionData(
            // method call here to change the chart data.
            getNum()
            //Math.random() * 100
        );
    }, 1000);
```

What we do here is: first, create the function for getting new data; second, define the variable(electionData) and asign it a value. This should be representative data, not necessarily correct for your case, but the data should work; third we update the electionData, or rather we set it again, giving electionData a completely new value. This should also work with arrays and other data types. 

If this doesn't help, let me know an I'll see if I can fix an example where the data in a line chart is cleared, and set again.

**Comment by rjstanford** [comment]
I may not have described it very well.  My expectation would be that if you used the line chart example and then called `self.lineChartData.removeAll()` that it would clear the data currently displayed. If you do that now then it breaks the chart.  I'm playing around with it (my workaround has been to remove and redisplay the chart for now), but didn't know if you had an 'aha!' about it.

**Comment by magnuskiro** [comment]
I just updated the third example, line-chat-view-model and e3.html. And I added an interval which removes all elements in the array. D3 fixes this just fine, it just transforms to the new data. 

The problem I found with the existing code was that in the html file we imported the wrong view-model, and in the view-model we should check to see if the lineChartData actually contains any data before connecting new with previous values. 

You might also find useful removeAll() stuff here: http://knockoutjs.com/documentation/observableArrays.html
